pkg/haprovider/haprovider.go

```diff
@@ -118,9 +118,16 @@ func (r *HAProvider) createService(
 			},
 		}
 	}
-	if ip, ok := cluster.ObjectMeta.Annotations[akoov1alpha1.ClusterControlPlaneAnnotations]; ok {
-		// "ip" can be ipv4 or hostname, add ipv4 or hostname to service.Spec.LoadBalancerIP
-		service.Spec.LoadBalancerIP = ip
+	if endpoint, ok := cluster.ObjectMeta.Annotations[akoov1alpha1.ClusterControlPlaneAnnotations]; ok {
+		// "endpoint" can be ipv4 or hostname, add ipv4 or hostname to service.Spec.LoadBalancerIP
+		if net.ParseIP(endpoint) == nil {
+			endpoint, err = queryFQDN(endpoint)
```

Comment:
would be even great to have a test covering this fallback logic?

Reply:
/lgtm

Reply:
I explored to add an integration test. It requires making an entry in /etc/hosts which is hard to achieve in our dev environment because of permission issues. I will add an e2e test later